bridge-proxy/src/bridge-proxy.rs

```diff
@@ -34,7 +34,7 @@ pub trait BridgeProxyContract:
 
     #[payable("*")]
     #[endpoint]
-    fn deposit(&self, eth_tx: EthTransaction<Self::Api>) {
+    fn deposit(&self, eth_tx: EthTransaction<Self::Api>, batch_id: u64) {
```

Comment:
I would not put this mandatory argument here, which is useless in the case we want to do a normal swap from Mvx->Eth
Instead, why not create a depositWithLinkedBatchID function? And that should be called from the bridge proxy SC